### ISSUE TITLE
Joined Channels REST API response too big/slow

### ISSUE THREAD

**Issue by luciofm** [RocketChat/Rocket.Chat#8891]
### Description:

Currently `/api/v1/channels.list.joined` will return an array with all joined usernames, and an array with muted usernames
This is a test user only joined on #general
```
curl -H "X-Auth-Token: xxxx" -H "X-User-Id: xxxx" https://demo.rocket.chat/api/v1/channels.list.joined
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 3357k    0 3357k    0     0   305k      0 --:--:--  0:00:11 --:--:--  607k
```
`channels.list` is also impacted by this and will return huge responses when returning big channels.

`groups.list` also has this problem, but usually groups are much smaller.

**Comment by jszaszvari** [comment]
How many users do you have? What is your setup?

Total response time with 2000 users on my instance is

198 ms for channels.list.joined
208 ms for channels.list
180 ms for groups.list

**Comment by graywolf336** [comment]
@jszaszvari Pulling the `open.rocket.chat` server's `#general` channel will result in this massive result where there are 158,479 users.

**Comment by graywolf336** [comment]
Will be resolved with #8947 and with a new message to developers about the change until `0.64` hits.
